src/rule-search.ts

```diff
@@ -0,0 +1,187 @@
+import * as vscode from 'vscode';
+import {window} from 'vscode';
+
+export class RuleInputProvider implements vscode.WebviewViewProvider {
+
+	public static readonly viewType = 'sourcery.rules';
+
+	private _view?: vscode.WebviewView;
+
+	private _extensionUri: vscode.Uri;
+
+	private _languageId: string;
+
+	constructor(
+		private _context: vscode.ExtensionContext,
+	) {
+		this._extensionUri = _context.extensionUri;
+	}
+
+	public async toggle() {
+		this._view.webview.postMessage({ command: 'toggle' });
+	}
+
+	public async resolveWebviewView(
+		webviewView: vscode.WebviewView,
+		context: vscode.WebviewViewResolveContext,
+		_token: vscode.CancellationToken,
+	) {
+		this._view = webviewView;
+
+		webviewView.webview.options = {
+			// Allow scripts in the webview
+			enableScripts: true,
+
+			localResourceRoots: [
+				this._extensionUri
+			]
+		};
+
+		this._languageId = this._resolveLanguage();
+		this._setTitle();
+
+		webviewView.webview.html = await this._getHtmlForWebview(webviewView.webview);
+
+		webviewView.webview.onDidReceiveMessage(async data => {
+			switch (data.type) {
+				case "scan": {
+					vscode.commands.executeCommand("sourcery.scan.rule", data.rule, data.advanced, false, this._languageId);
+					break;
+				}
+				case "replace": {
+					vscode.commands.executeCommand("sourcery.scan.rule", data.rule, data.advanced, true, this._languageId);
+					break;
+				}
+				case "save": {
+					window.showInformationMessage("Saving rules directly not yet available.")
+				}
+			}
+		});
+	}
+
+	private _setTitle() {
+		this._view.title = "Rules - " + this._languageId;
+	}
+
+	private _resolveLanguage() {
+		// Get the active editor in the workspace
+		const activeEditor = vscode.window.activeTextEditor;
+
+		// Check if there's an active editor
+		if (activeEditor) {
+			const languageId = activeEditor.document.languageId;
+			switch (languageId) {
+				case "python": {
+					return languageId;
+				}
+				case "typescript": {
+					return "javascript";
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** ‏This may produce surprising/incorrect results or just an error if typescript syntax is incorrect parsed by javascript